2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_queue/LinkedQueueTest.java

```diff
@@ -1,103 +1,382 @@
+        assertThat(sizeAfterPollOperation).isEqualTo(2);
+    }
 
-        integerQueue.poll(); // should poll 33
+    @Test
+    @Order(6)
+    void isEmptyWhenQueueIsEmpty() {
+        boolean isEmpty = this.integerQueue.isEmpty();
+        boolean isEmptyByReflection = isEmptyQueue();
 
-        assertEquals(123, integerQueue.poll().intValue());
-        assertEquals(2, integerQueue.size());
+        assertThat(isEmpty).isEqualTo(isEmptyByReflection);
     }
 
     @Test
     @Order(7)
-    void pollLastElement() {
-        integerQueue.add(8);
-        integerQueue.add(123);
-        integerQueue.add(99);
-        integerQueue.add(46);
+    void isEmptyAfterPollLastElement() {
+        addIntElementToQueue(1);
+        addIntElementToQueue(2);
 
-        integerQueue.poll(); // should poll 8
-        integerQueue.poll(); // should poll 123
-        integerQueue.poll(); // should poll 99
+        boolean isEmptyBeforePoll = isEmptyQueue();
+        pollElementFromQueue();
+        pollElementFromQueue();
+        boolean isEmptyAfterPoll = isEmptyQueue();
 
-        assertEquals(46, integerQueue.poll().intValue());
-        assertEquals(0, integerQueue.size());
+        assertThat(isEmptyBeforePoll).isEqualTo(false);
+        assertThat(isEmptyAfterPoll).isEqualTo(true);
     }
 
+
     @Test
     @Order(8)
-    void isEmptyOnEmptyQueue() {
-        assertTrue(integerQueue.isEmpty());
+    void pollFromEmptyQueue() {
+        boolean isEmpty = isEmptyQueue();
+        Integer firstElement = this.integerQueue.poll();
+
+        assertThat(isEmpty).isEqualTo(true);
+        assertThat(firstElement).isEqualTo(null);
     }
 
     @Test
     @Order(9)
-    void isEmpty() {
-        integerQueue.add(3);
-        integerQueue.add(9);
+    void pollFromNonEmptyQueue() {
+        addIntElementToQueue(11);
+        addIntElementToQueue(111);
+
+        boolean isEmpty = isEmptyQueue();
+        Integer firstElement = this.integerQueue.poll();
+
+        assertThat(isEmpty).isEqualTo(false);
+        assertThat(firstElement).isEqualTo(11);
+    }
+
+    @Test
+    @Order(10)
+    void pollLastElement() {
+        addIntElementToQueue(12);
+        addIntElementToQueue(10);
+
+        int sizeBeforePollOperation = getInternalSize();
+        boolean isEmptyBeforePoll = isEmptyQueue();
+        Integer firstElement = this.integerQueue.poll();
+        Integer secondElement = this.integerQueue.poll();
+        boolean isEmptyAfterPoll = isEmptyQueue();
+        int sizeAfterPollOperation = getInternalSize();
+
+        assertThat(isEmptyBeforePoll).isEqualTo(false);
+        assertThat(isEmptyAfterPoll).isEqualTo(true);
+
+        assertThat(sizeBeforePollOperation).isEqualTo(2);
+        assertThat(sizeAfterPollOperation).isEqualTo(0);
+
+        assertThat(firstElement).isEqualTo(12);
+        assertThat(secondElement).isEqualTo(10);
+    }
+
+
+    @Test
+    @Order(11)
+    void addElementIntoEmptyQueue() {
```

Comment:
Lets adhere to tests name convention like this: `methodNameDoesSomethingWhenSomeCondition`.
For instance `addFillsQueueWhenItISEmpty`